numpy/__init__.pyi

```diff
@@ -195,35 +195,24 @@ class flatiter:
     def __next__(self) -> Any: ...
 
 
-class ndarray(Iterable, Sized, SupportsInt, SupportsFloat, SupportsComplex,
-              SupportsBytes, SupportsAbs[Any]):
-
-    imag: ndarray
-    real: ndarray
-
+_ArraySelf = TypeVar("_ArraySelf", bound=_ArrayLike)
+class _ArrayLike(SupportsInt, SupportsFloat, SupportsComplex, SupportsBytes,
```

Comment:
Not a good name - in the docs, `array_like` means `can be converted to array`.

Perhaps `_ArrayOrScalarCommon`?